src/celeritas/em/msc/UrbanMscStepLimit.hh

```diff
@@ -164,53 +185,28 @@ UrbanMscStepLimit::UrbanMscStepLimit(UrbanMscRef const& shared,
 template<class Engine>
 CELER_FUNCTION auto UrbanMscStepLimit::operator()(Engine& rng) -> MscStep
 {
-    MscStep result;
-
-    result.phys_step = phys_step_;
-    result.true_path = phys_step_;
-
     // The case for a very small step or the lower limit for the linear
     // distance that e-/e+ can travel is far from the geometry boundary
-    // NOTE: use d_over_r_mh for muons and charged hadrons
-    if (result.true_path < shared_.params.limit_min_fix()
-        || range_ * msc_.d_over_r < safety_)
+    if (skip_displacement_)
     {
+        MscStep result;
         result.is_displaced = false;
-        auto temp = this->calc_geom_path(result.true_path);
-        result.geom_path = temp.geom_path;
-
+        result.true_path = phys_step_;
+        result.geom_path = this->calc_geom_path(phys_step_).geom_path;
         return result;
     }
 
     // Step limitation algorithm: UseSafety (the default)
-    // TODO: add options for other algorithms (see G4MscStepLimitType.hh)
-
-    // Initialisation at the first step or at the boundary
-    if (!msc_range_ || on_boundary_)
-    {
-        msc_range_.range_fact = params_.range_fact;
-        msc_range_.range_init = max<real_type>(range_, lambda_);
-        if (lambda_ > params_.lambda_limit)
-        {
-            msc_range_.range_fact
-                *= (real_type(0.75)
-                    + real_type(0.25) * lambda_ / params_.lambda_limit);
-        }
-        msc_range_.limit_min = this->calc_limit_min();
-
-        // Store persistent range properties within this tracking volume
-        physics_.msc_range(msc_range_);
-    }
-    // The step limit
     real_type limit = range_;
-    if (limit > safety_)
+    if (range_ > safety_)
```

Comment:
Why is this change meaningfully better than limit > safety ?

Reply:
The `limit` is the result of the calculation; the conditional here isn't "is the limit greater than the safety" but "is the range greater than the safety". The two are of course identical at this point but I think the second conveys the intended meaning (and since the range doesn't change, it's arguably less error-prone).